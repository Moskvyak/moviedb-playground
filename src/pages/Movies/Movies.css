.movies__grid-container {
    display: grid;
    grid-template-columns: 50% 50%;
    align-content: space-around;
}

.movies__list-container {
    margin-top: 20px;
}
.movies__list-container h3 {
    color: #FFF;
    margin: 20px;
    margin-bottom: 0;
}

.grid-item {
    font-size: 100%;
    margin: 10px 20px;
    text-align: center;
    position: relative;
}
.movies__header {
    width: 100%;
    background-image: url("../../background.png");
    background-size: 100% auto;
}
.movies__logo-wrapper {
    padding: 40px 0px;
    width: 60px;
    background: #081c24;
    margin: auto;
}

@media screen and (min-width: 600px) {
    .movies__grid-container {
        display: grid;
        grid-template-columns: 33% 33% 33%;
        align-content: space-around;
    }
    .movies__header {
        height: 250px;
    }
    .movies__logo-wrapper {
        width: 100px;
        z-index: 1;
        padding: 50px 0px;
    }
}

@media screen and (min-width: 960px) {
    .movies__grid-container {
        display: grid;
        grid-template-columns: 25% 25% 25% 25%;
        align-content: space-around;
    }
}

@media screen and (min-width: 1280px) {
    .movies__grid-container {
        display: grid;
        grid-template-columns: 20% 20% 20% 20% 20%;
        align-content: space-around;
    }
}