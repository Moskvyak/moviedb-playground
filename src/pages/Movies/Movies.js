import React from 'react';

import Logo from '../../components/Logo';
import SearchForm from '../../components/SearchForm';
import MovieListItem from '../../components/MovieListItem';
import { searchMovies } from '../../services/Dataservice';

import './Movies.css';

const MIN_CHARACTERS_FOR_SEARCH = 1;

class Movies extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      movies: [],
      searchValue: ''
    };
  }

  componentDidMount = async () => {
    this.fetchResults();
  };

  fetchResults = async () => {
    try {
      const { searchValue } = this.state;
      const movies = await searchMovies(searchValue);
      this.setState({ movies });
    } catch (error) {
      this.setState({ error: 'Server error' });
    }
  };
  clearSearch = () => {
    this.setState({ searchValue: ''}, this.fetchResults);
  }
  onSearchValueChange = event => {
    const { target: { value } } = event;
    if (!value) {
      this.clearSearch();
      return;
    }

    this.setState({ searchValue: value }, () => {
      if (value.length >= MIN_CHARACTERS_FOR_SEARCH) {
        this.fetchResults();
      }
    });
  };

  render() {
    const { movies, searchValue } = this.state;
    return (
      <div className="page-background-color page-wrapper primary-color">
        <div className="movies__header">
          <div className="movies__logo-wrapper">
            <Logo />
          </div>
          <SearchForm
            error={false}
            value={searchValue}
            loading={false}
            onChange={this.onSearchValueChange}
            clearSearch={this.clearSearch}
            placeholder="Search"
          />
        </div>
        <div className="movies__list-container">
            <h3>Popular movies</h3>
            <div className="movies__grid-container">
            {movies.map(movie => <MovieListItem key={movie.id} { ...movie }/>)}
            </div>
        </div>
      </div>
    );
  }
}
export default Movies;
