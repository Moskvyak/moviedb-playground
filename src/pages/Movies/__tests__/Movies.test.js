import React from 'react';
import renderer from 'react-test-renderer';
import Movies from '../Movies';

describe('<Movies />', () => {
  it('should render with required props', () => {
    // TODO overriding ComponentDidMount, better solution would be to use mock data
    Movies.componentDidMount = () => {}
    const component = renderer.create(<Movies />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
