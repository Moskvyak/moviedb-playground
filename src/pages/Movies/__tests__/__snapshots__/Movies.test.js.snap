// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Movies /> should render with required props 1`] = `
<div
  className="page-background-color page-wrapper primary-color"
>
  <div
    className="movies__header"
  >
    <div
      className="movies__logo-wrapper"
    >
      <img
        alt="Logo"
        src="movie-db-logo.svg"
      />
    </div>
    <div
      className="searchform__wrapper primary-color"
    >
      <input
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Search"
        type="text"
        value=""
      />
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-search fa-w-16 fa-1x searchform__right-icon"
        color="#01d277"
        data-icon="search"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <div
    className="movies__list-container"
  >
    <h3>
      Popular movies
    </h3>
    <div
      className="movies__grid-container"
    />
  </div>
</div>
`;
